### ISSUE TITLE
Unexpected error "feature 'ref struct interfaces' is not available"

### ISSUE THREAD

**Issue by jjonescz** [dotnet/roslyn#76401]
**Version Used**: 4.13.0-3.24612.1 ([c30de30](https://github.com/dotnet/roslyn/commit/c30de30c96df7a9310ecaaa317a2745036a74f5c))

**Steps to Reproduce**:

Compile with LangVersion=12:

```cs
using System.Diagnostics.CodeAnalysis;
interface I
{
    void M();
}
struct S : I
{
    [UnscopedRef] void I.M() { } // errors here
}
```

**Expected Behavior**: Only error CS9102: UnscopedRefAttribute cannot be applied to an interface implementation because implemented member 'I.M()' doesn't have this attribute.

**Actual Behavior**: Additional error CS9202: Feature 'ref struct interfaces' is not available in C# 12.0. Please use language version 13.0 or greater.

But there is no ref struct implementing an interface.

**Comment by AlekseyTs** [comment]
C# 12 disallows applying  UnscopedRefAttribute to an interface implementation regardless whether the implemented interface member has UnscopedRefAttribute. The "Ref Struct Interfaces" feature allowed applying the attribute to such a member in a struct (https://github.com/dotnet/csharplang/blob/main/proposals/csharp-13.0/ref-struct-interfaces.md#ref-struct-interfaces-1):

> When a `struct` / `ref struct` member implements an interface member with a `[UnscopedRef]` attribute, the implementing member may also be decorated with `[UnscopedRef]` but it is not required.

 Hence the language feature check (regardless whether the implementing struct is a ref struct) and the additional error.
